### ISSUE TITLE
Strange failure when building stack via cabal on ghc 7.8 - travis disabled

### ISSUE THREAD

**Issue by mgsloan** [commercialhaskell/stack#2190]
Strange failure on https://travis-ci.org/commercialhaskell/stack/jobs/132728622 .  Opening this issue as I'm adding it to "allowed failures"

```
store-0.1.0.1 depends on th-lift-instances-0.1.7 which failed to install.
th-lift-instances-0.1.7 failed during the configure step. The exception was:
user error
(/opt/ghc/7.8.4/lib/ghc-7.8.4/unix-2.7.0.1/libHSunix-2.7.0.1.a(HsUnix.o): In
function `__hsunix_ptsname':
(.text+0x120): multiple definition of `__hsunix_ptsname'
/home/travis/.cabal/lib/x86_64-linux-ghc-7.8.4/unix-2.7.2.0/libHSunix-2.7.2.0.a(HsUnix.o):HsUnix.c:(.text+0x20):
first defined here
/opt/ghc/7.8.4/lib/ghc-7.8.4/unix-2.7.0.1/libHSunix-2.7.0.1.a(HsUnix.o): In
function `__hsunix_grantpt':
(.text+0x130): multiple definition of `__hsunix_grantpt'
/home/travis/.cabal/lib/x86_64-linux-ghc-7.8.4/unix-2.7.2.0/libHSunix-2.7.2.0.a(HsUnix.o):HsUnix.c:(.text+0x30):
first defined here
/opt/ghc/7.8.4/lib/ghc-7.8.4/unix-2.7.0.1/libHSunix-2.7.0.1.a(HsUnix.o): In
function `__hsunix_unlockpt':
(.text+0x140): multiple definition of `__hsunix_unlockpt'
/home/travis/.cabal/lib/x86_64-linux-ghc-7.8.4/unix-2.7.2.0/libHSunix-2.7.2.0.a(HsUnix.o):HsUnix.c:(.text+0x40):
first defined here
/opt/ghc/7.8.4/lib/ghc-7.8.4/unix-2.7.0.1/libHSunix-2.7.0.1.a(HsUnix.o): In
function `__hsunix_push_module':
(.text+0x150): multiple definition of `__hsunix_push_module'
/home/travis/.cabal/lib/x86_64-linux-ghc-7.8.4/unix-2.7.2.0/libHSunix-2.7.2.0.a(HsUnix.o):HsUnix.c:(.text+0x50):
first defined here
collect2: ld returned 1 exit status
)
```

**Comment by mgsloan** [comment]
This particular failure seems to have resolved itself: https://travis-ci.org/commercialhaskell/stack/builds/134067096

**Comment by borsboom** [comment]
I've hit this now too: https://travis-ci.org/commercialhaskell/stack/jobs/210240124
Things seem to be getting confused between the version of `unix` that's included with GHC (2.7.1.0 in this case) and the version being used for the build (2.7.2.1).

**Comment by borsboom** [comment]
FYI: Upgrading to cabal-install-1.24 seems to have resolved my problem.
